### PR TITLE
Replace algorithm for triangulation from n-gon in IGenerateProxy

### PR DESCRIPTION
**Pull request by ZarothYe**
Triangulate convex n-gons with standard fan triangulation. Now with less gnashing of teeth, fewer f-bombs, and (probably) more efficiency!

See:
https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxguide/Manual/Geometry.html#id2
https://en.wikipedia.org/wiki/Fan_triangulation

### CODE REVIEW

File: Sources/Plasma/PubUtilLib/plPhysX/plPXPhysical.cpp [hunk 1]
```diff
@@ -627,28 +628,16 @@ static plDrawableSpans* IGenerateProxy(plDrawableSpans* drawable,
...
-                ptr = indices;
-                do {
-                    tris.push_back(centerIdx);
-                    tris.push_back(*ptr++);
-                    tris.push_back(ptr < end ? *ptr : *indices);
-                } while (ptr < end);
+                // Use simple fan triangulation because this is known to be a convex polygon
+                // https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxguide/Manual/Geometry.html#id2
+                for (physx::PxU16 j = 2; j < polygon.mNbVerts; j++)
+                {
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plPhysX/plPXPhysical.cpp:634]:
```suggestion
                for (physx::PxU16 j = 2; j < polygon.mNbVerts; j++) {
```

;)

Comment (ZarothYe) [reply Sources/Plasma/PubUtilLib/plPhysX/plPXPhysical.cpp:634]: Dangit, I always forget that rule. Thanks!

**Comment by Hoikas** [comment]
Hmm... it looks like a spurious merge commit [f2cd0c2](https://github.com/H-uru/Plasma/pull/1323/commits/f2cd0c26fd3cdc2ba32504f0d2415e4b0ac1074b) was introduced 🤔

**Comment by ZarothYe** [comment]
> Hmm... it looks like an errant merge commit [f2cd0c2](https://github.com/H-uru/Plasma/pull/1323/commits/f2cd0c26fd3cdc2ba32504f0d2415e4b0ac1074b) was introduced 🤔

Sorry, didn't check my git log before pushing. Rebased now to remove the merge commit.

**Review by Hoikas — APPROVED** [review]

**Pull request merged**